README.md

```diff
@@ -1,145 +1,308 @@
+* `-instances="project1:region:instance1,project3:region:instance1"`
+
+A comma-separated list of instances to open inside `-dir`. Also supports
+exposing a TCP port and renaming the default Unix Domain Sockets; see examples
+below.  Same list can be provided via INSTANCES environment variable, in case
+when both are provided - proxy will use command line flag.
+
+**Example**
+
+Using TCP sockets:
+
+```
+./cloud_sql_proxy -instances=my-project:us-central1:sql-inst=tcp:3306 &
+mysql -u root -h 127.0.0.1
+```
+
+Using Unix sockets:
+
+```
+./cloud_sql_proxy -dir=/cloudsql -instances=my-project:us-central1:sql-inst &
+mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+```
+
+To specify a custom Unix socket name:
+
+```
+./cloud_sql_proxy -dir=/cloudsql \
+    -instances=my-project:us-central1:sql-inst=unix:custom_socket_name &
+mysql -u root -S /cloudsql/custom_socket_name
+```
+
+To specify a custom location for a Unix socket (overrides `-dir`):
+
+```
+./cloud_sql_proxy -dir=/cloudsql \
+    -instances=my-project:us-central1:sql-inst=unix:/my/custom/sql-socket &
+mysql -u root -S /my/custom/sql-socket
+```
+
+* `-fuse`
+
+Requires access to `/dev/fuse` as well as the `fusermount` binary. An optional
+`-fuse_tmp` flag can specify where to place temporary files. The directory
+indicated by `-dir` is mounted.
+
+**Example**
+
+Using `-fuse`, you do not need to specify instance names ahead of time:
+
+```
+./cloud_sql_proxy -dir=/cloudsql -fuse &
+mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+```
+
+* `-instances_metadata=metadata_key`
+
+Usable on [GCE](https://cloud.google.com/compute/docs/quickstart) only. The
+given [GCE metadata](https://cloud.google.com/compute/docs/metadata) key will be
+polled for a list of instances to open in `-dir`. The metadata key is relative
+from `computeMetadata/v1/`. The format for the value is the same as the
+'instances' flag. A hanging-poll strategy is used, meaning that changes to the
+metadata value will be reflected in the `-dir` even while the proxy is running.
+When an instance is removed from the list the corresponding socket will be
+removed from `-dir` as well (unless it was also specified in `-instances`), but
+any existing connections to this instance will NOT be terminated.
+
+**Example**
+
+```
+./cloud_sql_proxy -dir=/cloudsql \
+    -instances_metadata instance/attributes/<custom-metadata-key> &
+mysql -u root -S /cloudsql/my-project:us-central1:sql-inst
+```
 
 Note: `-instances` and `-instances_metadata` may be used at the same time but
 are not compatible with the `-fuse` flag.
 
-cloud_sql_proxy authentication can be configured in a few different ways. Those listed higher on the list will override options lower on the list:
+* `-max_connections`
+
+If provided, the maximum number of connections to establish before refusing new
+connections. Defaults to 0 (no limit). Note: requests made through the
+SQL Admin API are rate limited to 2x the number of connections per minute.
```

Comment:
> Note: requests made through the SQL Admin API are rate limited to 2x the number of connections per minute.

We should probably just remove this line - this is incorrect:
1. The total number of _refresh attempts_ is limited to 2 time per minute _per instance_
2. There is additional retry logic _per request_. Currently each request can be tried up to 5x times (this may change soon)

Reply:
For now I've removed the note. Future passes can add this detail back in, especially if (or when) it changes.